Array_Vector_Exercises/Exercise1/TDinamicArray.h

```diff
@@ -0,0 +1,242 @@
+		return *this;
+	}
+
+	//----------------------------------------------------------------------
+	T& operator[] (const int Index)
+	{
+		return DynamicArray[Index];
+	}
+
+	const T& operator[] (const int Index) const
+	{
+		return DynamicArray[Index];
+	}
+
+	//-----------------------------------------------------------------------
+	T& Front()
+	{
+		return DynamicArray[0];
+	}
+
+	const T& Front() const
+	{
+		return DynamicArray[0];
+	}
+
+	//------------------------------------------------------------------------
+	T& Back()
+	{
+		return DynamicArray[SizeArray - 1];
+	}
+
+	const T& Back() const
+	{
+		return DynamicArray[SizeArray - 1];
+	}
+
+	//-----------------------------------------------------------------------
+	T* GetData()
+	{
+		return DynamicArray;
+	}
+
+	const T* GetData() const
+	{
+		return DynamicArray;
+	}
+
+	//------------------------------------------------------------------------
+	bool IsEmpty() const
+	{
+		return SizeArray == 0;
+	};
+
+	//-------------------------------------------------------------------------
+	int GetSize() const
+	{
+		return SizeArray;
+	}
+
+	//-------------------------------------------------------------------------
+	int GetCapacity() const
+	{
+		return CapacityArray;
+	}
+
+	//-------------------------------------------------------------------------
+	void Reserve(const int NewCapacity)
+	{
+		if (NewCapacity > CapacityArray)
+		{
+			T* ContainerArray = new T[NewCapacity];
+
+			for (int i = 0; i < SizeArray; i++)
+			{
+				ContainerArray[i] = DynamicArray[i];
+			}
+			delete[] DynamicArray;
+
+			DynamicArray = ContainerArray;
+			CapacityArray = NewCapacity;
+		}
+	}
+
+	//--------------------------------------------------------------------------
+	void ShrinkToFit()
+	{
+		if (CapacityArray != SizeArray)
+		{
+			T* ContainerArray = new T[SizeArray];
+
+			for (int i = 0; i < SizeArray; i++)
+			{
+				ContainerArray[i] = DynamicArray[i];
+			}
+			delete[] DynamicArray;
+
+			DynamicArray = ContainerArray;
+		}
+	}
+
+	//--------------------------------------------------------------------------
+	void Clear()
+	{
+		SizeArray = 0;
+	}
+
+	//--------------------------------------------------------------------------
+	void PushBack(const T& InType)
+	{
+		if (SizeArray == CapacityArray)
+		{
+			Reserve(CapacityArray * 2);
+			DynamicArray[SizeArray] = InType;
+			SizeArray++;
+		}
+		else
+		{
+			DynamicArray[SizeArray] = InType;
+			SizeArray++;
+		}
+	}
+
+	//-----------------------------------------------------------------------------
+	void PopBack()
+	{
+		if (SizeArray != 0)
+		{
+			SizeArray--;
+		}
+	}
+
+	//-----------------------------------------------------------------------------
+	void Insert(const int Index, const T& InType)
+	{
+		if (Index < SizeArray && Index >= 0)
+		{
+			if (SizeArray == CapacityArray)
+			{
+				Reserve(CapacityArray * 2);
+			}
+				
+			for (int i = SizeArray; i > Index; i--)
+			{
+				DynamicArray[Index] = DynamicArray[i - 1];
```

Comment:
This doesn't seem like it would work. You changing the same value every iteration.